private/model/api/passes.go

```diff
@@ -244,6 +244,46 @@ func (a *API) renameExportable() {
 	}
 }
 
+// renameCollidingFields will rename any fields that uses an SDK or Golang
+// specific name.
+func (a *API) renameCollidingFields() {
+	for _, v := range a.Shapes {
+		namesWithSet := map[string]struct{}{}
+		for k, field := range v.MemberRefs {
+			if strings.HasPrefix(k, "Set") {
+				namesWithSet[k] = struct{}{}
+			}
+
+			if collides(k) {
+				delete(v.MemberRefs, k)
+				v.MemberRefs[k+"_"] = field
+			}
+		}
+
+		// checks if any field names collide with setters.
+		for name := range namesWithSet {
+			setterName := fmt.Sprintf("Set%s", name)
+			if field, ok := v.MemberRefs[setterName]; ok {
+				delete(v.MemberRefs, name)
+				v.MemberRefs[name+"_"] = field
+			}
+		}
+	}
+
+}
+
+// collides will return true if it is a name used by the SDK or Golang.
+func collides(name string) bool {
```

Comment:
Would be helpful if this function name was a little bit more specific. Collides is a little to generic.  Something like `IsGeneratedMethod`.  Let the decision on what to do with the fact a name is a generated method left up to the user of the function.